File: 104-maximum-depth-of-binary-tree/maximum-depth-of-binary-tree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def maxDepth(self, root: Optional[TreeNode]) -> int:
        if root != None:
            count_right = self.maxDepth(root.right) if root.right else 0
            count_left = self.maxDepth(root.left) if root.left else 0
            return 1 + max(count_left, count_right)
        else:
            return 0

File: 104-maximum-depth-of-binary-tree/maximum-depth-of-binary-tree.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    int maxDepth(TreeNode* root) {

       if (!root) return 0;

       int leftMax = 1 + maxDepth(root->right);
       int rightMax = 1 + maxDepth(root->left);

       if (leftMax > rightMax) return leftMax;
        else return rightMax;
        // int rightMax = 1
        // int leftMax = 1

        // if (root.left) leftMax++;
        // if (root.right) rightMaxx++:

        // max = 
    }
};